allen-cell-animated/website-3d-cell-viewer
# feat: Replaced Dockerfile

Closes #169, "Dockerfile is out of date". Adds a new Dockerfile that performs the build and makes the files available via a nginx dev server, and fixes some other bugs along the way with video.

## Changes
- Deletes old Dockerfile and related config files.
- Adds a new Dockerfile that builds the project, then serves the files over http using an nginx server.
- Updates webpack config to fix a bug where videos would be missing during builds. (This *should* also fix the missing banner video on GitHub pages.)

## Verification
Follow the steps specified in the README! Spelling them out here:
1. Install Docker locally: [instructions for Windows](https://docs.docker.com/desktop/install/windows-install/), [instructions for Mac](https://docs.docker.com/desktop/install/mac-install/)
2. Clone the project and navigate to the root.
3. Run `docker build -t 3d-volume-viewer .`
4. Run `docker run --rm -p 8080:80 --name 3d-volume-viewer 3d-volume-viewer`
5. Go to http://localhost:8080. Pages should load in as expected.
6. Refreshing any of the subpages should still show the subpage as expected.
